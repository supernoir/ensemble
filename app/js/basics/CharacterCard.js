import React from 'react';
import { Link } from 'react-router-dom';
import { Card, Icon, Button } from 'semantic-ui-react';
import PropTypes from 'prop-types';
import intl from 'react-intl-universal';

/**
 * Class CharacterCard
 * Displays Characters in a Card View
 */
export default class CharacterCard extends React.Component {
	componentDidMount() {
		this.props.getProjectTitle(this.props.character.project);
	}

	render() {
		return (
			<Card>
				<Card.Content>
					<Card.Header>
						<Link to={`/character/${this.props.character._id}`}>{this.props.character.full_name}</Link>
					</Card.Header>

					{this.props.character.birthday !== void 0 && this.props.character.birthday.length > 0
						? <Card.Meta>
							<Icon name={'birthday cake'} /> {this.props.character.birthday}
						</Card.Meta>
						: null}

					{this.props.character.gender !== void 0 && this.props.character.gender.length > 0
						? <Card.Meta>
							<Icon name={'transgender'} /> {this.props.character.gender}
						</Card.Meta>
						: null}

					<Card.Description>{this.props.character.desc}</Card.Description>
				</Card.Content>
				{this.props.character.project !== void 0 && this.props.projectTitle !== void 0
					? <Card.Content>
						<Button icon basic labelPosition="left">
							<Icon name={'book'} /> <Link to={`/project/${this.props.character.project}`}>{this.props.projectTitle}</Link>
						</Button>
					</Card.Content>
					: null}
				<Card.Content extra>
					<Link to={`/editcharacter/${this.props.character._id}`}>
						<Button circular basic icon="edit" color="blue" content={intl.get('app.edit')} />
					</Link>
					<Button circular basic icon="delete" color="red" content={intl.get('app.delete')} />
				</Card.Content>
			</Card>
		);
	}
}

CharacterCard.propTypes = {
	projectTitle   : PropTypes.string,
	getProjectTitle: PropTypes.func,
	character      : PropTypes.shape({
		_id      : PropTypes.Number,
		project  : PropTypes.Number,
		desc     : PropTypes.string,
		gender   : PropTypes.string,
		birthday : PropTypes.string,
		full_name: PropTypes.string
	})
};
