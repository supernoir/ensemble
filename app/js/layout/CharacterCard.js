import React from 'react';
import { Link } from 'react-router-dom';
import { Card, Icon, Button } from 'semantic-ui-react';

export default class CharacterCard extends React.Component {
	componentDidMount() {
		this.props.getProjectTitle(this.props.character.project);
	}

	render() {
		return (
			<Card>
				<Card.Content>
					<Card.Header>
						<Link to={`/character/${this.props.character._id}`}>{this.props.character.full_name}</Link>
					</Card.Header>
					<Card.Meta>
						<Icon name={'birthday cake'} /> {this.props.character.birthday}
					</Card.Meta>
					<Card.Meta>
						<Icon name={'transgender alternate'} /> {this.props.character.gender}
					</Card.Meta>
					<Card.Description>{this.props.character.desc}</Card.Description>
				</Card.Content>
				{this.props.character.project !== void 0 ? (
					<Card.Content>
						<Icon name={'book'} /> <Link to={`/project/${this.props.projectTitle}`}></Link>
					</Card.Content>
				) : null}
				<Card.Content extra>
					<Link to={`/editcharacter/${this.props.character._id}`}>
						<Button circular icon="edit" color="green" />
					</Link>
					<Button circular icon="delete" color="red" />
				</Card.Content>
			</Card>
		);
	}
}
