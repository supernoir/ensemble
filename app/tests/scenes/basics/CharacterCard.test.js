import React from 'react';
import { shallow } from 'enzyme';

const testData = {
	projectTitle   : 'TestTitle',
	getProjectTitle: jest.fn(),
	character      : {
		_id      : 1,
		project  : 2,
		desc     : 'TestDescription',
		gender   : 'TestGender',
		birthday : '01/01/1990',
		full_name: 'TestName'
	}
};

// Import Components
import CharacterCard from '../../../js/basics/CharacterCard';

test('CharacterCard -- Snapshot Test', () => {
	const wrapper = shallow(<CharacterCard
		getProjectTitle={() => testData.getProjectTitle()}
		character={testData.character}
		projectTitle={testData.projectTitle}
	/>
	);
	expect(wrapper).toMatchSnapshot();
});
