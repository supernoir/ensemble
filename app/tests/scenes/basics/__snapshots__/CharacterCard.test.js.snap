// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard -- Snapshot Test 1`] = `
<Card>
  <CardContent>
    <CardHeader>
      <Link
        replace={false}
        to="/character/1"
      >
        TestName
      </Link>
    </CardHeader>
    <CardMeta>
      <Icon
        as="i"
        name="birthday cake"
      />
       
      01/01/1990
    </CardMeta>
    <CardMeta>
      <Icon
        as="i"
        name="transgender"
      />
       
      TestGender
    </CardMeta>
    <CardDescription>
      TestDescription
    </CardDescription>
  </CardContent>
  <CardContent>
    <Button
      as="button"
      basic={true}
      icon={true}
      labelPosition="left"
    >
      <Icon
        as="i"
        name="book"
      />
       
      <Link
        replace={false}
        to="/project/2"
      >
        TestTitle
      </Link>
    </Button>
  </CardContent>
  <CardContent
    extra={true}
  >
    <Link
      replace={false}
      to="/editcharacter/1"
    >
      <Button
        as="button"
        basic={true}
        circular={true}
        color="blue"
        content=""
        icon="edit"
      />
    </Link>
    <Button
      as="button"
      basic={true}
      circular={true}
      color="red"
      content=""
      icon="delete"
    />
  </CardContent>
</Card>
`;
